src/formatDate.js

```diff
@@ -8,7 +8,42 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat.pop();
+  const toSeparator = toFormat.pop();
+  const dateParts = date.split(fromSeparator);
+
+  let year, month, day;
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    if (fromFormat[i] === 'YYYY') {
+      year = dateParts[i];
+    } else if (fromFormat[i] === 'YY') {
+      year =
+        parseInt(dateParts[i]) < 30 ? '20' + dateParts[i] : '19' + dateParts[i];
```

Comment:
The logic for handling two-digit years assumes that any year less than 30 belongs to the 2000s and any other belongs to the 1900s. This might not be accurate for all use cases. Consider making this logic more flexible or configurable.